### PR TITLE
Remove "Demo" from example titles

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

In the context of example titles "Demo" is a fill word without meaning. All examples are demos.

This PR removes some usages of "Demo". If possible, the title is changed to something more meaningful. (Which is also the reason, I'm only doing part of the examples. Good titles need a bit of thought for every single case).

### CODE REVIEW

File: examples/axes_grid1/demo_axes_rgb.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """
-=============
-Demo Axes RGB
-=============
+=======
+RGBAxes
```
Comment (story645) [inline examples/axes_grid1/demo_axes_rgb.py:3]: Not sure this is all that clear either - what about RGB composite image or RGB channels or even Toolkit RGB Axes/Toolkit RGB Axes helper

**Review by story645 — APPROVED** [review]

File: examples/axes_grid1/demo_axes_rgb.py (hunk 1 above)
Comment (timhoffm) [reply examples/axes_grid1/demo_axes_rgb.py]: Changed to "Showing RGB channels using RGBAxes" plus description update.

**Comment by story645** [comment]
Not sure why all the doc builds decided to fail 😕 (otherwise would merge)

**Comment by timhoffm** [comment]
Docs fixed.

For info: This was failing:
~~~
/home/circleci/project/doc/gallery/axes_grid1/demo_axes_rgb.rst:16: WARNING: more than one target found for cross-reference 'RGBAxes': mpl_toolkits.axes_grid1.axes_rgb.RGBAxes, mpl_toolkits.axisartist.axes_rgb.RGBAxes
~~~

As a general rule, if you want to find out, what made the doc builds fail:

1. Click on the failed build (e.g. docs-python38)
2. Download the full log.
3. Search case-sensitive for "WARNING" in the full log.

95% of the time that shows you why sphinx was unhappy.

**Pull request merged**